controller/src/test/java/io/pravega/controller/util/RetryHelperTest.java

```diff
@@ -107,14 +107,14 @@ public void testLoopWithDelay() {
         AtomicInteger loopCounter = new AtomicInteger();
         ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
 
-        AtomicLong previous = new AtomicLong(System.currentTimeMillis());
+        AtomicLong previous = new AtomicLong(System.nanoTime());
         AtomicBoolean loopDelayHonored = new AtomicBoolean(true);
 
         long oneSecond = Duration.ofSeconds(1).toMillis();
         RetryHelper.loopWithDelay(
                 () -> loopCounter.incrementAndGet() < maxLoops,
                 () -> {
-                    loopDelayHonored.compareAndSet(true, System.currentTimeMillis() - previous.get() > oneSecond);
+                    loopDelayHonored.compareAndSet(true, System.nanoTime() - previous.get() >= oneSecond);
```

Comment:
I think this is incorrect because `System.nanoTime() - previous.get()` is giving us a value in nanoseconds, while `oneSecond` is computed in milliseconds.

Reply:
Earlier the precision was in milli seconds. And we were doing strictly greater than one second and that could cause test to fail sometime. 

What we are doing here is increasing the precision of difference from milli to nano seconds. 
Note: Its a diff of two nano times. So it should give us a value of One second or more with precision of nano second.

Reply:
`oneSecond` is still in milliseconds, see line 113.

Reply:
got it.. thanks